### PR TITLE
Feature/info card

### PR DESCRIPTION
**Pull request by jgfidelis**
Example of something that could be used in the storecomponents :P
https://card2--storecomponents.myvtex.com/

wks with example of many uses and different props:
https://card3--storecomponents.myvtex.com/
On this wks:
```
"info-card#1": {
    "props": {
      "isFullModeStyle": true,
      "textPosition": "left",
      "headline": "Enjoy pure water throughout your home",
      "subline": "Your family relies on clean water to stay healthy. Choose from a high-quality selection of water softeners and filtration systems to remove water contaminants and hardness in your home. Take control with a water system designed around your unique needs.",
      "callToActionMode": "button",
      "callToActionText": "Shop Water Filtration",
      "callToActionUrl": "s/under-sink-filtration",
      "imageUrl": "https://ecowaterqa.vteximg.com.br/arquivos/ids/155513",
      "mobileImageUrl": "https://ecowaterqa.vteximg.com.br/arquivos/ids/155513",
      "textAlignment": "left"
    }
  },
  "info-card#2": {
    "props": {
      "isFullModeStyle": false,
      "textPosition": "left",
      "headline": "Enjoy pure water throughout your home",
      "subline": "Your family relies on clean water to stay healthy. Choose from a high-quality selection of water softeners and filtration systems to remove water contaminants and hardness in your home. Take control with a water system designed around your unique needs.",
      "callToActionMode": "button",
      "callToActionText": "Shop Water Filtration",
      "callToActionUrl": "s/under-sink-filtration",
      "imageUrl": "https://ecowaterqa.vteximg.com.br/arquivos/ids/155513",
      "mobileImageUrl": "https://ecowaterqa.vteximg.com.br/arquivos/ids/155513",
      "textAlignment": "left"
    }
  },
  "info-card#3": {
    "props": {
      "isFullModeStyle": true,
      "textPosition": "center",
      "headline": "Enjoy pure water throughout your home",
      "subline": "Your family relies on clean water to stay healthy. Choose from a high-quality selection of water softeners and filtration systems to remove water contaminants and hardness in your home. Take control with a water system designed around your unique needs.",
      "callToActionMode": "link",
      "callToActionText": "Shop Water Filtration",
      "callToActionUrl": "s/under-sink-filtration",
      "imageUrl": "https://ecowaterqa.vteximg.com.br/arquivos/ids/155513",
      "mobileImageUrl": "https://ecowaterqa.vteximg.com.br/arquivos/ids/155513",
      "textAlignment": "left"
    }
  },
  "info-card#4": {
    "props": {
      "isFullModeStyle": false,
      "textPosition": "right",
      "headline": "Enjoy pure water throughout your home",
      "subline": "Your family relies on clean water to stay healthy. Choose from a high-quality selection of water softeners and filtration systems to remove water contaminants and hardness in your home. Take control with a water system designed around your unique needs.",
      "callToActionMode": "link",
      "callToActionText": "Shop Water Filtration",
      "callToActionUrl": "s/under-sink-filtration",
      "imageUrl": "https://images.pexels.com/photos/207301/pexels-photo-207301.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
      "mobileImageUrl": "https://images.pexels.com/photos/207301/pexels-photo-207301.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
      "textAlignment": "right"
    }
  },
```

TODO: add basic Readme

#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/InfoCard/CallToAction.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+const ActionWrapper = ({ mode, text }) => {
+  if (mode === callActionValues.BUTTON) {
+    return (
+      <Button size="small" onClick={() => {}}>{text}</Button>
+    )
+  }
+
+  // Mode is link
+  return <p className="link t-body b underline c-action-primary">{text}</p>
```
Comment (lbebber) [inline react/components/InfoCard/CallToAction.js]: shouldn't this be an actual link?

File: react/components/InfoCard/CallToAction.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import { Button } from 'vtex.styleguide'
+import { Link } from 'vtex.render-runtime'
+import styles from './infoCard.css'
+
+import { callActionValues } from './SchemaTypes'
+
+const ActionWrapper = ({ mode, text }) => {
+  if (mode === callActionValues.BUTTON) {
+    return (
+      <Button size="small" onClick={() => {}}>{text}</Button>
```
Comment (lbebber) [inline react/components/InfoCard/CallToAction.js]: shouldn't the button receive an `onClick` handler?

File: react/components/InfoCard/index.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+  [textPostionValues.LEFT]: 'flex-row',
+  [textPostionValues.RIGHT]: 'flex-row-reverse'
+}
+
+const defaultValues = {
+  textPosition: textPositionTypes.TEXT_POSITION_LEFT.value,
+  textAlignment: textAlignmentTypes.TEXT_ALIGNMENT_LEFT.value,
+}
+
+const getEnumValues = enumObject => Object.values(enumObject).map(({ value }) => value)
```
Comment (lbebber) [inline react/components/InfoCard/index.js]: :pick: the browser compatibility of `Object.values` is a bit weird (IE has no support whatsoever and only a relatively recent safari version has it), perhaps use the `values` function from `ramda`, which does the same thing, or adapt the function to use `Object.keys`

File: react/components/InfoCard/index.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        enumNames: getEnumNames(textPositionTypes),
+        default: textPositionTypes.TEXT_POSITION_LEFT.value,
+      },
+      headline: {
+        title: 'editor.info-card.headline.title',
+        description: 'editor.info-card.headline.description',
+        type: 'string',
+        default: null,
+      },
+      subline: {
```
Comment (lbebber) [inline react/components/InfoCard/index.js]: the text below a headline is called a `subhead`. I'm not sure "subline" exists, but I might be mistaken

File: react/components/InfoCard/index.js [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+  textAlignment: textAlignmentTypes.TEXT_ALIGNMENT_LEFT.value,
+}
+
+const getEnumValues = enumObject => Object.values(enumObject).map(({ value }) => value)
+const getEnumNames = enumObject => Object.values(enumObject).map(({ name }) => name)
+
+const safelyGetToken = (tokenMap, valueWanted, propName) => tokenMap[valueWanted] || defaultValues[propName]
+
+class InfoCard extends PureComponent {
+  static propTypes = {
```
Comment (matheusps) [inline react/components/InfoCard/index.js]: Personally, I think this looks better outside the component.

File: react/components/InfoCard/index.js [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+    subline: string,
+    callToActionMode: oneOf(getEnumValues(callToActionModeTypes)),
+    callToActionText: string,
+    callToActionUrl: string,
+    imageUrl: string,
+    mobileImageUrl: string,
+    textAlignment: oneOf(getEnumValues(textAlignmentTypes)),
+  }
+
+  static defaultProps = {
```
Comment (matheusps) [inline react/components/InfoCard/index.js]: This one too

File: react/components/InfoCard/index.js [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+    subline: null,
+    callToActionMode: callToActionModeTypes.CALL_ACTION_BUTTON.value,
+    callToActionText: '',
+    callToActionUrl: '',
+    imageUrl: '',
+    mobileImageUrl: null,
+    textAlignment: textAlignmentTypes.TEXT_ALIGNMENT_LEFT.value,
+  }
+
+  static schema = {
```
Comment (matheusps) [inline react/components/InfoCard/index.js]: Also :)

File: react/components/InfoCard/index.js [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+  textPosition: textPositionTypes.TEXT_POSITION_LEFT.value,
+  textAlignment: textAlignmentTypes.TEXT_ALIGNMENT_LEFT.value,
+}
+
+const getEnumValues = enumObject => Object.values(enumObject).map(({ value }) => value)
+const getEnumNames = enumObject => Object.values(enumObject).map(({ name }) => name)
+
+const safelyGetToken = (tokenMap, valueWanted, propName) => tokenMap[valueWanted] || defaultValues[propName]
+
+class InfoCard extends PureComponent {
```
Comment (matheusps) [inline react/components/InfoCard/index.js]: Why a class if you do not use states?

File: react/components/InfoCard/index.js [hunk 9]
```diff
@@ -0,0 +1,195 @@
+import React, { PureComponent } from 'react'
+import { bool, string, oneOf } from 'prop-types'
+import classNames from 'classnames'
+import { withRuntimeContext } from 'vtex.render-runtime'
```
Comment (matheusps) [inline react/components/InfoCard/index.js]: If you change to a functional component, you could try `useRuntime` hook. Your tree would look cleaner

File: react/components/InfoCard/index.js (hunk 5 above)
Comment (lbebber) [reply react/components/InfoCard/index.js]: I disagree hahah (just my 2c though)

File: react/components/InfoCard/CallToAction.js (hunk 1 above)
Comment (jgfidelis) [reply react/components/InfoCard/CallToAction.js]: Its wrapped in a `<Link>`

File: react/components/InfoCard/index.js (hunk 4 above)
Comment (jgfidelis) [reply react/components/InfoCard/index.js]: done

File: react/components/InfoCard/index.js (hunk 3 above)
Comment (jgfidelis) [reply react/components/InfoCard/index.js]: done

**Pull request merged**